libraries/AP_HAL/UARTDriver.cpp

```diff
@@ -224,3 +224,15 @@ void AP_HAL::UARTDriver::log_stats(const uint8_t inst, StatsTracker &stats, cons
 }
 #endif // HAL_LOGGING_ENABLED
 #endif // HAL_UART_STATS_ENABLED
+
+const extern AP_HAL::HAL& hal;
+
+extern "C" {
+    void hal_console_printf(const char *, ...);
+}
+void hal_console_printf(const char *fmt, ...) {
+    va_list ap;
```

Comment:
we already do this with __wrap_printf(), why another copy?

Reply:
This is indeed unnecessary, I tested and bare `printf` works fine in the LittleFS code because of that hook. Could revert commit a39ad0681094c2f2923aac0108be8bf0d4eef1e3.

Reply:
Dropped